Andy anthony

andyfiveeleven opened the pull request:
#### Single-line Summary
**Today, Anthony and Andy paired together. It took about 2 hours**

#### Reflect and summarize on your process for each `TODO` item :  
  1. First, we started on the basic script tags and adding a data attribute, that was fairly straightforward
  2. Next, we worked on teh filtering, that went well, but we forgot to concatinate $(this).val() the first time, again though fairly easy
  3.  The handle main nav was a bit complex, but knowing about .data selector was CLUTCH
  4. finally i learned some things about hidden things. 


#### Checklist (before submitting, fill in each set of square brackets with an 'x')
- [x] We have titled the Pull Request similar to our branch name (ex: 'brian-rick'). 
- [x] This PR includes commits from both myself and my partner; e.g. We followed good pair programming practices by switching driver/navigator roles.
- [x] There is no extraneous, unrelated code included in this PR.
- [x] We have summarized our `TODO:` process above.

andyfiveeleven committed:
author and category filtering enabled

brainspork committed:
completes handle main nav and sets teasers then calls prototypes

andyfiveeleven committed:
fixed the script tag, made it all render

andyfiveeleven committed:
better styling

cathy810218 commented on starter-code/scripts/articleView.js:
```diff
@@ -33,28 +29,26 @@ articleView.populateFilters = function() {
+      $('article').hide();
+      $('article[data-category="'+ $(this).val() +'"]').show();
+    } else {
+      $('article').show();
```
OH NO! You're also showing the template :(((

cathy810218 commented on starter-code/scripts/articleView.js:
```diff
@@ -33,28 +29,26 @@ articleView.populateFilters = function() {
-      // TODO: If the select box was changed to an option that is blank, we should
-      //       show all the articles, except the one article we are using as a template.
-
+      $('article').show();
```
You are showing all articles including the template.

andyfiveeleven replied:
WHOOPS! I forgot my :not(.template) ! Anthony's computer crashed right when we were finishing up, and we thought we'd caught all the changes, but didn't click back into the re-show. Should I resubmit the whole thing?

cathy810218 replied:
You don't have to resubmit the whole thing. Just push another commit and submit the link again on canvas :)

cathy810218 commented on starter-code/scripts/articleView.js:
```diff
@@ -33,28 +29,26 @@ articleView.populateFilters = function() {
-      //       Use an "attribute selector" to find those articles, and fade them in for the reader.
-
+      $('article').hide();
+      $('article[data-author="'+ $(this).val() +'"]').show();
```
`show()` works, but don't forget that we can also use `fadeIn()`.
> Use an "attribute selector" to find those articles, and fade them in for the reader.

cathy810218 commented on starter-code/scripts/articleView.js:
```diff
@@ -64,7 +58,11 @@ articleView.handleMainNav = function() {
   //       So: You need to dynamically build a selector string with the correct ID, based on the
   //       data available to you on the .tab element that was clicked.
-
+  $('.main-nav ul').on('click', 'li', function(){
```
If you read the `TODO` comment again, you will notice that it asks you to use `.tab` element.
i.e `$('.main-nav').on('click', '.tab', function(){`

If now you click on `Github` and watch what happens on your page.
In the html file, we have
````
<li class="tab" data-content="articles"><a href="#" class="icon-home"> Home</a></li>
<li class="tab" data-content="about"><a href="#" class="icon-address-book"> About</a></li>
<li><a href="http://twitter.com/brookr" target="_blank" class="icon-twitter3"> Twitter</a></li>
<li><a href="http://github.com/brookr" target="_blank" class="icon-github"> GitHub</a></li>
````
both Twitter and GitHub don't have `data-content`.
And that's why we only want to handle event when the element class is `tab`.

cathy810218 commented on starter-code/scripts/articleView.js:
```diff
@@ -80,9 +78,18 @@ articleView.setTeasers = function() {
+  $('#articles').on('click', '.read-on', function(hash){
+    hash.preventDefault();
+    $(this).parent().find(':hidden').show();
+  });
```
Read on button should be hidden once it's clicked as it's stated in TODO

brainspork committed:
Fixes template display bug

Adds :not(.template) selector to handleAuthorFilter to stop the display of the template
